Split between using sdf and using gazebo

doudou opened the pull request.

doudou committed:
syskit: add world code to allow managing a Gazebo instance from Cucumber steps

doudou committed:
syskit: refactor to use the unmanaged task support

doudou committed:
syskit: allow setting the world to load in Syskit's configuration object

This allows to override the world from the CLI using

  --set gazebo.world=world_name

doudou committed:
syskit: allow starting the gazebo instance in a separate working directory

This is to avoid polluting the current directory

doudou committed:
syskit: demote the message about the world_file_path override to info

If the user does not realize that

  --set gazebo.world_file_path

overrides the path, then I don't know ...

doudou committed:
fix logging

doudou committed:
fix the corba scheduler warnings

doudou committed:
properly deinitialize the CORBA stuff on destruction

doudou committed:
syskit: properly pass the world_name argument from use_gazebo_world to world_from_path

doudou committed:
syskit: fix setup in simulation mode

doudou committed:
ruby: set the diff filter for ruby files

doudou committed:
syskit: tell syskit whether the gazebo instance runs locally or not

This allows to optimize if the Syskit app is configured to use e.g.
MQueues

doudou committed:
syskit: fix initialization with localhost: false

doudou committed:
syskit: start a visualization if CUCUMBER_VIZ is set to 1

doudou committed:
syskit: refactor between sdf and gazebo usage

The "*_sdf_*" methods now allow to use SDF models outside gazebo,
i.e. assume that the SDF information represents known information
about the vehicle's world.

The "_*gazebo_*" methods on the other hand assume that the goal is
to interface with a gazebo instance, and define the relevant information
(such as e.g. devices)

doudou committed:
syskit: annotate devices with frame information

doudou committed:
syskit: make the world be 'world'

There can be only one world in the syskit context, and there's no
point in making our life miserable like this.

doudou committed:
ruby: must use 'world' as frame name for all worlds in visualization setup as well

Otherwise 'viz' won't work for the Syskit-setup gazebos

doudou committed:
allow to only install the Ruby bindings

This allows to use the Syskit-SDF functionality without having to
install the simulation.

The better solution would be to split the package between the C++
plugin, the Syskit-SDF plugin and the Rock-Gazebo Ruby integration,
but that's for another time.

doudou committed:
syskit: factor out the code necessary to create a link export

This is a big chunk of code, and we've already copy/pasted it twice.

doudou committed:
viz: set the vizkit3d plugin name to the link/model name

dbcesar commented:
@doudou, I can't neither run `syskit test -rlive` nor `syskit ide -rgazebo`. cause the following is

`Roby[WARN]: undefined method 'use_sdf_model' for Profile:FlatFish::Profiles::Live::Base:Syskit::Actions::Profile (NoMethodError)`

Then I `require 'rock_gazebo/syskit` and do like profiles/gazebo/base.rb `use_sdf_model: 'model://flat_fish/model.sdf'` and then I'm having this on `profiles/live/base.rb`

```
live[WARN]: while calling #<Proc:0x00000004946d68@/home/diego/dev.bir/bundles/flat_fish/config/robots/live.rb:22>
Roby[WARN]: expected the XML element to be a 'sdf' tag, but got <sdf><world name='world'/></sdf> (ArgumentError)
Roby[WARN]: = Backtrace
Roby[WARN]: = 
Roby[WARN]: 
Roby[WARN]:   /home/diego/dev.bir/control/ruby_sdformat/lib/sdf/element.rb:36:in `initialize'
Roby[WARN]:   /home/diego/dev.bir/control/ruby_sdformat/lib/sdf/world.rb:16:in `new'
Roby[WARN]:   /home/diego/dev.bir/control/ruby_sdformat/lib/sdf/world.rb:16:in `empty'
Roby[WARN]:   /home/diego/dev.bir/install/lib/ruby/2.3.0/rock_gazebo/syskit/profile_extension.rb:6:in `sdf_world'
Roby[WARN]:   /home/diego/dev.bir/install/lib/ruby/2.3.0/rock_gazebo/syskit/profile_extension.rb:54:in `use_sdf_world'
Roby[WARN]:   /home/diego/dev.bir/bundles/flat_fish/models/profiles/live/base.rb:48:in `block in <module:Live>'
Roby[WARN]:   /home/diego/dev.bir/tools/syskit/lib/syskit/actions/profile.rb:519:in `instance_eval'
Roby[WARN]:   /home/diego/dev.bir/tools/syskit/lib/syskit/actions/profile.rb:519:in `profile'
Roby[WARN]:   /home/diego/dev.bir/bundles/flat_fish/models/profiles/live/base.rb:47:in `<module:Live>'
...
```

doudou commented:
Weird ... I have another problem.

The sdf issue is fixed on `control/ruby_sdformat` ... was sitting on my machine.

doudou committed:
syskit: do not inject all models and links as transform producers implicitly

The few devices / requirements that need it must now call

  transformer_uses_sdf_links_of(model_dev)

explicitly

doudou committed:
ignore vim swapfiles

doudou committed:
syskit: accept model://model_name as a shortcut for model://model_name/model.sdf

doudou committed:
viz: the expected world frame is now 'world'

doudou committed:
in setup_scene, modify the transformer configuration to introduce 'world' in-between the models and the actual world name

Otherwise the OSG tree gets utterly broken when the configuration is

   world -> actual_world_name
   actual_world_name -> object

but it is fed the world -> object transformation

doudou committed:
syskit: allow using model://flat_fish as a shortcut for model://flat_fish/model.sdf in Profile#use_sdf_model

doudou committed:
syskit: store the SDF element on the device instances
